packages/graphql-hooks/src/middlewares/apqMiddleware.js

```diff
@@ -0,0 +1,39 @@
+async function sha256(data) {
+  // MOCK
+  return data
+    .split('')
+    .reduce((acc, ch) => acc + ch.charCodeAt(0), '')
+    .substr(0, 8)
+}
+
+/**
+ * AutomaticPersistedQueryMiddleware - must be last in the middleware list
+ * @param {function} makeRequest
+ * @returns Promise<object>
+ */
+const APQMiddleware = async ({ operation, client, resolve, reject }, next) => {
+  try {
+    // Add SHA256 of a query
+    operation = { ...operation, hash: await sha256(operation.query) }
+
+    // Try to send just the hash
+    const res = await client.requestViaHttp(operation, {
+      fetchOptionsOverrides: { method: 'GET' },
+      hashOnly: true
+    })
+
+    if (!res.error) resolve(res)
```

Comment:
I'd avoid inline ifs for consistency (always seeing curly braces for JS flow constructs) and readability. Just a suggestion.